src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -130,19 +136,37 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
         else:
             logger.warning(UPGRADE_MSG)
     if exit_code:
-        telemetry.set_failure("CLI upgrade failed.")
+        err_msg = "CLI upgrade failed."
+        logger.warning(err_msg)
+        telemetry.set_failure(err_msg)
         sys.exit(exit_code)
-    # Updating Azure CLI and extension together is not supported in homebrewe package.
+
+    import azure.cli.core
+    import importlib
+    importlib.reload(azure.cli.core)
+    new_version = azure.cli.core.__version__
+
+    if update_cli and new_version == local_version:
```

Comment:
Why not compare `new_version` and `latest_version` to determine whether the upgrade is successful?

Reply:
To avoid issues caused by the case that a new version is released during user's upgrade epspecially for unplanned hot fix release, so the `latest_version` retrieved earlier may be smaller than the actual updated `new_version`, which is a false alarm here.

Reply:
Got it